test/models/fraction.spec.ts

```diff
@@ -149,6 +149,19 @@ describe("Fraction", () => {
       const f = new Fraction(tenPow18, 1);
       assert.equal(f.toNumber(), 1e18);
     });
+
+    it("Can serialize number larger than float", () => {
+      const f = new Fraction(
+        new BN(
+          "4572703011319588167347432415546880036279866516943345249514864772813530139562604616171825435245554855143198721150328379973878658975213121347107927848905622759808528046680694931071812505099249835494785625055292121740875930346662225138972256888437500357739010418759122146937662234675889914922526355031952562363843368434796080505266339127627357626793894148384690286306732645932753887586933120444885403006477593485191718007262006965294247514114763095506587377085460750400000"
+        ),
+        new BN(
+          "4774642967403412778558002718245820511160340393936581384568251929433121266847718907194950001102609642111137823519729464988698100962023057203306804424578581032780993445591560733837592299028888352218568642023539602725628897211541316901852536439083302862262101219849809407717422630910964615478205410140280934609846041106236330315316634515479595683742854529339310209222679458248292443955431255834882063370342436913436854307979328247288503275306278914502165428008437070219920998172584805926041217"
+        )
+      );
+      assert.isNotNaN(f.toNumber());
+      assert.isNotNaN(f.inverted().toNumber());
+    });
```

Comment:
Could we add test cases for very large numerator with a small denominator and very large denominator with a small numerator as well?